### DISCUSSION TITLE
enodebd Hard-code IP Address

### DISCUSSION THREAD

**Discussion by parthiban337** [magma/magma#7936] [Q&A]
Magma AGW Version 1.6.0 

enodebd Logs:

● magma@enodebd.service - Magma enodebd service
     Loaded: loaded (/etc/systemd/system/magma@.service; disabled; vendor preset: enabled)
     Active: active (running) since Thu 2021-07-08 06:27:28 UTC; 14min ago
   Main PID: 142390 (python3)
      Tasks: 9 (limit: 9263)
     Memory: 53.7M (limit: 300.0M)
     CGroup: /system.slice/system-magma.slice/magma@enodebd.service
             └─142390 python3 -m magma.enodebd.main
Jul 08 06:27:28 wavelabstest systemd[1]: Started Magma enodebd service.
Jul 08 06:27:28 wavelabstest enodebd[142390]: ERROR:root:Error retrieving config for control_proxy, key not found: sentry_url_python
Jul 08 06:27:28 wavelabstest enodebd[142390]: INFO:root:Starting enodebd...
Jul 08 06:27:28 wavelabstest enodebd[142390]: INFO:spyne.application:Initializing application {urn:dslforum-org:cwmp-1-0}Tr069Application...
Jul 08 06:27:28 wavelabstest enodebd[142390]: INFO:root:Listening on address 127.0.0.1:60055
Jul 08 06:27:28 wavelabstest enodebd[142390]: INFO:root:Starting TR-069 server on 172.16.5.11:48080
Jul 08 06:27:28 wavelabstest enodebd[142390]: WARNING:root:The IP address of the eth1 interface is 172.16.5.11. The expected IP addresses are ('192.168.60.142', '10.0.2.1')
Jul 08 06:27:28 wavelabstest sudo[142402]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/usr/sbin/iptables -t nat -L
Jul 08 06:27:28 wavelabstest sudo[142402]: pam_unix(sudo:session): session opened for user root by (uid=0)
Jul 08 06:27:29 wavelabstest sudo[142402]: pam_unix(sudo:session): session closed for user root

I backtrace the source code now... and found below
​this IP Address what we see as an error of the enodebd is a hard-coded value 
​    
in the lte\gateway\python\magma\enodebd\enodebd_iptables_rules.py
​    line number 140 
​    
Here it has boolean to verify  verify_config=does_iface_config_match_expected(
​    and in the same file in the top line number 51 for the boolean function and in line number 38 the values

EXPECTED_IP4 = ('192.168.60.142', '10.0.2.1')
EXPECTED_MASK = '255.255.255.0'

This looks to be a bug here...
Anyone from enodebd team, can you please check & comment on this?

**Comment by amarpad** [comment]
cc @andreilee

**Comment by ardzoht** [comment]
Is this affecting the applied iptables rules or the functionality of enodebd service configuring your eNBs? I agree there is some hardcoded values for expected IP which don't apply to all cases, but from the `enodebd_iptables_rules` scope it seems `verify_config` displays a warning if the eth1 interface IP is not aligned and it continues with the configuration. I'm interested on the impact of this beyond the warning.

**Reply by parthiban337** [reply]
Tried testing from SRS simulator, S1 connection is failing & I am getting the above logs in the AGW terminal.... This is not allowing to establish S1 connections.
